From 1946b536d75fc47f16bf3accb9b18c98145280b1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Blin?=
 <sebastien.blin@savoirfairelinux.com>
Date: Tue, 17 Apr 2018 09:47:47 -0400
Subject: [PATCH] settings: check if g_get_user_special_dir fails

g_get_user_special_dir can return a nullptr, so the string
construction will fails badly. Must be checked or fallback to
G_USER_DIRECTORY_DESKTOP ($HOME/Desktop)

Change-Id: I04b72221f14091db2cfd880a9b35b8ab92e46ae4
Reviewed-by: Philippe Gorley <philippe.gorley@savoirfairelinux.com>
---
 src/generalsettingsview.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/generalsettingsview.cpp b/src/generalsettingsview.cpp
index 4c6a15a..4fa2d9c 100644
--- a/src/generalsettingsview.cpp
+++ b/src/generalsettingsview.cpp
@@ -248,7 +248,11 @@ general_settings_view_init(GeneralSettingsView *self)
     auto* download_directory_variant = g_settings_get_value(priv->settings, "download-folder");
     char* download_directory_value;
     g_variant_get(download_directory_variant, "&s", &download_directory_value);
-    std::string default_download_dir = g_get_user_special_dir (G_USER_DIRECTORY_DOWNLOAD);
+    std::string default_download_dir = {};
+    if (auto* directory = g_get_user_special_dir (G_USER_DIRECTORY_DOWNLOAD))
+        default_download_dir = directory;
+    else
+        default_download_dir = g_get_user_special_dir (G_USER_DIRECTORY_DESKTOP);
     auto current_value = std::string(download_directory_value);
     if (current_value.empty()) {
         g_settings_set_value(priv->settings, "download-folder", g_variant_new("s", default_download_dir.c_str()));
-- 
2.17.0

